eng/DownloadSharedSource.ps1

```diff
@@ -2,29 +2,54 @@
+$armCoreSharedPath = "sdk/resourcemanager/Azure.ResourceManager/src/Shared/"
+
+if(Test-Path $clonedPath)
+{
+    Remove-Item $clonedPath -Recurse -Force
+}
+
+Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/Azure/azure-sdk-for-net.git $clonedPath"
+git clone --no-checkout --filter=tree:0 https://github.com/Azure/azure-sdk-for-net.git $clonedPath
+Push-Location $clonedPath
+Write-Host "git sparse-checkout init"
+git sparse-checkout init
+Write-Host "git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath"
+git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath
+Write-Host "git checkout"
+git checkout
+Pop-Location
+
+function CopyAll([string[]]$files, [string]$source, [string]$destination)
 {
     foreach ($file in $files)
     {
-        Write-Host "Downloading" $file
-        $text = (Invoke-WebRequest -Uri "$baseUrl/$file").Content
-        $text.Trim() | Out-File (Join-Path $downloadPath $file)
+        Write-Host "Copying $file to $destination"
+        Copy-Item (Join-Path $source $file) (Join-Path $destination $file)
     }
 }
 
-$downloadPath = Resolve-Path (Join-Path $PSScriptRoot '..' 'src' 'assets' 'Azure.Core.Shared')
-Get-ChildItem $downloadPath -Filter *.cs | Remove-Item;
 $files = @('AsyncLockWithValue.cs', 'ClientDiagnostics.cs', 'DiagnosticScope.cs', 'DiagnosticScopeFactory.cs', 'ContentTypeUtilities.cs', 'HttpMessageSanitizer.cs',
     'OperationInternalBase.cs', 'OperationInternal.cs', 'OperationInternalOfT.cs', 'TaskExtensions.cs', 'Argument.cs', 'Multipart/MultipartFormDataContent.cs',
     'Multipart/MultipartContent.cs', 'AzureKeyCredentialPolicy.cs', 'AppContextSwitchHelper.cs',
     'OperationPoller.cs', 'FixedDelayWithNoJitterStrategy.cs', 'SequentialDelayStrategy.cs',
     'ForwardsClientCallsAttribute.cs', 'AsyncLockWithValue.cs', 'VoidValue.cs')
-$baseUrl = 'https://raw.githubusercontent.com/Azure/azure-sdk-for-net/main/sdk/core/Azure.Core/src/Shared/'
-DownloadAll $files $baseUrl $downloadPath
+$sourcePath = "$clonedPath/sdk/core/Azure.Core/src/Shared/"
+$destinationPath = "$repoRoot/src/assets/Azure.Core.Shared"
+
+Get-ChildItem $destinationPath -Filter *.cs | Remove-Item;
+CopyAll $files $sourcePath $destinationPath
 
 #Download management Shared
 $files = 'SharedExtensions.cs', 'ManagedServiceIdentityTypeV3Converter.cs'
-$downloadPath = Resolve-Path (Join-Path $PSScriptRoot '..' 'src' 'assets' 'Management.Shared')
-Get-ChildItem $downloadPath -Filter *.cs | Remove-Item;
-$baseUrl = 'https://raw.githubusercontent.com/Azure/azure-sdk-for-net/main/sdk/resourcemanager/Azure.ResourceManager/src/Shared/'
-DownloadAll $files $baseUrl $downloadPath
+$sourcePath = "$clonedPath/sdk/resourcemanager/Azure.ResourceManager/src/Shared"
+$destinationPath = "$repoRoot/src/assets/Management.Shared"
+
+Get-ChildItem $destinationPath -Filter *.cs | Remove-Item;
+CopyAll $files $sourcePath $destinationPath
+
+Start-Sleep -Seconds 1
```

Comment:
What is this sleep for?

Reply:
Alson wondering about this sleep.

Reply:
The folder can be locked for delete. I used "-ErrorAction SilentlyContinue" to prevent an error from bubbling when this occors, but it only happens when there was recent io action on the folder.  I'm thinking it's probable Windows Defender scanning the files.

I would have just left the folder in place and skipped the delete, but this script is expected to be run on dev machines (presumably Windows with Defender).  The 1 second sleep appears to be enough time to let all the other consumers release the file.

Reply:
OK please add a comment in the code to that affect otherwise everyone that looks at this will wonder the same thing and likely end up deleting it.